needle-slack-connector/src/app/_components/slack/SlackWorkspacePreview.tsx

```diff
@@ -0,0 +1,215 @@
+        // Decreasing number of messages per channel: 5, 3, 2, 1, 1
+        const messagesToTake =
+          index === 0 ? 5 : index === 1 ? 3 : index === 2 ? 2 : 1;
+        messagesToShow.push(...channelMessages.slice(0, messagesToTake));
+      });
+
+    return messagesToShow;
+  };
+
+  if (isLoading || !workspace) {
+    return (
+      <div className="my-8 flex flex-col">
+        <div className="mt-4 text-center text-zinc-500">
+          Loading workspace resources...
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="my-8 space-y-6">
+      <div className="rounded-lg border border-zinc-200 p-4 dark:border-zinc-700">
+        <div className="flex items-center gap-4">
+          <img
+            src={workspace.team.icon?.image_68 ?? "/icons/slack.svg"}
+            alt={workspace.team.name}
+            className="h-12 w-12 rounded-lg"
+          />
+          <div>
+            <h2 className="text-xl font-semibold">{workspace.team.name}</h2>
+            <p className="text-sm text-zinc-500">
+              {workspace.team.domain}.slack.com
+            </p>
+          </div>
+        </div>
+
+        <div className="mt-6 space-y-2">
+          <div className="mb-4 flex items-center justify-between">
+            <h3 className="text-lg font-medium">
+              Showing {channels?.length ?? 0} Channels
+            </h3>
+          </div>
+
+          {!channels?.length ? (
+            <div className="text-sm text-zinc-500">No channels available</div>
+          ) : (
+            <div className="space-y-2">
+              {channels.map((channel) => {
+                // Get only canvas tabs count
+                const canvasCount =
+                  channel.properties?.tabs?.filter(
+                    (tab) => tab.type === "canvas" && tab.data?.file_id,
+                  ).length ?? 0;
+
+                return (
+                  <div
+                    key={channel.id}
+                    onClick={() => handleChannelToggle(channel)}
+                    className="flex cursor-pointer items-center rounded-md border border-zinc-200 p-4 hover:bg-zinc-50 dark:border-zinc-700 dark:hover:bg-zinc-900"
+                  >
+                    <input
+                      type="checkbox"
+                      checked={selectedChannels.some(
+                        (c) => c.id === channel.id,
+                      )}
+                      className="mr-4 rounded border-gray-300"
+                      onClick={(e) => e.stopPropagation()}
+                      onChange={() => handleChannelToggle(channel)}
+                    />
+                    <div className="flex flex-1 items-center justify-between">
+                      <div className="space-y-1">
+                        <p className="font-medium">#{channel.name}</p>
+                        {canvasCount > 0 && (
+                          <div className="flex items-center gap-1">
+                            <svg
```

Comment:
Let's factor out SVG - will make it more readable.

Reply:
Good thinking, replacing it with releted icon from `lucide-react`